src/main/java/genetics/ApiInstance.java

```diff
@@ -0,0 +1,117 @@
+package genetics;
+
+import com.google.common.base.Preconditions;
+
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import genetics.api.IGeneticApiInstance;
+import genetics.api.IGeneticFactory;
+import genetics.api.IGeneticSaveHandler;
+import genetics.api.alleles.IAlleleHelper;
+import genetics.api.alleles.IAlleleRegistry;
+import genetics.api.individual.IChromosomeList;
+import genetics.api.individual.IKaryotype;
+import genetics.api.root.IIndividualRoot;
+import genetics.api.root.IIndividualRootHelper;
+import genetics.api.root.IRootDefinition;
+import genetics.api.root.components.IRootComponentRegistry;
+
+import genetics.alleles.AlleleHelper;
+import genetics.alleles.AlleleRegistry;
+import genetics.classification.ClassificationRegistry;
+import genetics.individual.ChromosomeList;
+import genetics.individual.GeneticSaveHandler;
+import genetics.individual.RootDefinition;
+import genetics.root.IndividualRootHelper;
+import genetics.root.RootComponentRegistry;
+
+public enum ApiInstance implements IGeneticApiInstance {
+	INSTANCE;
+
+	private static final String ERROR_MESSAGE = "A method of the genetic api was called before the api reached the state at that the value of the method is present.";
+
+	@Nullable
+	public ClassificationRegistry classificationRegistry;
+	@Nullable
+	public AlleleRegistry alleleRegistry;
+
+	private final Map<String, RootDefinition> rootDefinitionByUID = new HashMap<>();
+	private final Map<String, IChromosomeList> chromosomeListByUID = new HashMap<>();
+
+	@Override
+	public ClassificationRegistry getClassificationRegistry() {
+		Preconditions.checkState(classificationRegistry != null, ERROR_MESSAGE);
+		return classificationRegistry;
+	}
+
+	public void setClassificationRegistry(@Nullable ClassificationRegistry classificationRegistry) {
+		this.classificationRegistry = classificationRegistry;
+	}
+
+	@Override
+	public IAlleleRegistry getAlleleRegistry() {
```

Comment:
How will methods like this interact with parallel mod loading? Not sure this one will be too bad but I think there may be race conditions in other places